operator/deploy/crd.yaml

```diff
@@ -30,11 +30,335 @@ spec:
+                    annotations:
+                      type: object
+                      properties:
+                        api_spec_annotation_name:
+                          type: string
+                          default: kiali.io/api-spec
+                        api_type_annotation_name:
+                          type: string
+                          default: kiali.io/api-type
+                auth:
+                  type: object
+                  properties:
+                    strategy:
+                      type: string
+                      enum:
+                        - login
+                        - openshift
+                        - anonymous
+                    ldap:
+                      type: object
+                      properties:
+                        ldap_base:
+                          type: string
+                        ldap_bind_dn:
+                          type: string
+                        ldap_group_filter:
+                          type: string
+                        ldap_host:
+                          type: string
+                        ldap_insecure_skip_verify:
+                          type: boolean
+                          default: false
+                        ldap_mail_id_key:
+                          type: string
+                          default: "mail"
+                        ldap_member_of_key:
+                          type: string
+                          default: "memberOf"
+                        ldap_port:
+                          type: number
+                        ldap_role_filter:
+                          type: string
+                        ldap_search_filter:
+                          type: string
+                        ldap_use_ssl:
+                          type: boolean
+                        ldap_user_filter:
+                          type: string
+                        ldap_user_id_key:
+                          type: string
+                          default: "cn"
+                deployment:
+                  type: object
+                  properties:
+                    accessible_namespaces:
+                      type: array
+                      items:
+                        type: string
+                    additional_service_yaml:
+                      type: object
+                      additionalProperties: true
+                    affinity:
+                      type: object
+                      properties:
+                        affinity:
+                          type: object
+                          additionalProperties: true
+                        pod:
+                          type: object
+                          additionalProperties: true
+                        pod_anti:
+                          type: object
+                          additionalProperties: true
+                    image_name:
+                      type: string
+                    image_pull_policy:
```

Comment:
if we ARE going to put defaults in this schema, then we need to set this default which is "IfNotPresent"